consensus/safety-rules/src/safety_rules.rs

```diff
@@ -82,13 +82,31 @@ impl<T: Payload> SafetyRules<T> {
         }
     }
 
-    pub fn signer(&self) -> &ValidatorSigner {
-        &self.validator_signer
+    /// This verifies a QC makes sense in the current context, specifically that this is for the
+    /// current epoch and extends from the preffered round.
+    fn verify_qc(&self, qc: &QuorumCert) -> Result<(), Error> {
+        let validator_verifier = self
+            .validator_verifier
+            .as_ref()
+            .ok_or(Error::NotInitialized)?;
+
+        qc.verify(validator_verifier)
+            .map_err(|e| Error::InvalidQuorumCertificate(e.to_string()))?;
+
+        if qc.parent_block().round() < self.persistent_storage.preferred_round()? {
+            Err(Error::InvalidQuorumCertificate(
```

Comment:
this shouldn't be an error, just ignore it otherwise caller has to ensure this

Reply:
Shouldn't the caller?

Reply:
This shouldn't have any ramifications as is, so landing without modification. Will adjust the caller to differentiate liveness, safety, and warning "errors"